main/src/ca/uwaterloo/flix/api/lsp/provider/completion/Completion.scala

```diff
@@ -80,6 +81,43 @@ sealed trait Completion {
     case Completion.VarCompletion(sym, tpe, context, flix) =>
       CompletionItem(label = sym.text, sortText = Priority.local(sym.text), textEdit = TextEdit(context.range, sym.text),
         detail = Some(FormatType.formatType(tpe)(flix)), kind = CompletionItemKind.Variable)
+    case Completion.DefCompletion(decl, context, flix) =>
+      val name = decl.sym.toString
+      CompletionProvider.getApplySnippet(name, decl.spec.fparams)(context).map(snippet => {
+        CompletionItem(label = CompletionProvider.getLabelForNameAndSpec(decl.sym.toString, decl.spec)(flix),
+          sortText = Priority.normal(name),
+          filterText = Some(CompletionProvider.getFilterTextForName(name)),
+          textEdit = TextEdit(context.range, snippet),
+          detail = Some(FormatScheme.formatScheme(decl.spec.declaredScheme)(flix)),
+          documentation = Some(decl.spec.doc.text),
+          insertTextFormat = InsertTextFormat.Snippet,
+          kind = CompletionItemKind.Function)
+      }).orNull
```

Comment:
Wait. You can't return `null` here. You must never return `null`.

Reply:
Maybe you have to refactor `getApplySnippet` into two functions: One that checks the number of arguments and one that does its computation.

Reply:
In that case, what if the required number of arguments isn't provided, what should be returned then?

Reply:
Then the completion does not apply, right?

That's how the old code works, right?

Reply:
True.

Hmm, hopefully I'll come up with a neat way to do it and get rid of the `Option`.